integration/kubernetes/confidential/image_pulling_with_snapshotter.bats

```diff
@@ -0,0 +1,155 @@
+#	RUNTIMECLASS: set the runtimeClassName value from $RUNTIMECLASS.
+#
+new_pod_config() {
+    local base_config="${FIXTURES_DIR}/pod-config.yaml.in"
+    local image="$1"
+    local index="$2"
+
+    local new_config=$(mktemp "${BATS_FILE_TMPDIR}/$(basename ${base_config}).XXX")
+    IMAGE="$image" RUNTIMECLASS="$RUNTIMECLASS" INDEX="$2" envsubst <"$base_config" >"$new_config"
+    echo "$new_config"
+}
+
+setup() {
+    start_date=$(date +"%Y-%m-%d %H:%M:%S")
+
+    pod_config="$(new_pod_config "$image_simple_signed")"
+    pod_id=""
+
+    kubernetes_delete_all_cc_pods_if_any_exists || true
+
+    echo "Prepare containerd for Confidential Container"
+    SAVED_CONTAINERD_CONF_FILE="/etc/containerd/config.toml.$$"
+    configure_cc_containerd "$SAVED_CONTAINERD_CONF_FILE"
+
+    echo "Reconfigure Kata Containers"
+    switch_image_service_offload off
+    clear_kernel_params
+    add_kernel_params "${original_kernel_params}"
+
+    setup_proxy
+    switch_measured_rootfs_verity_scheme none
+}
+
+# Check the logged messages on host have a given message.
+# Parameters:
+#      $1 - the message
+#
+# Note: get the logs since the global $start_date.
+#
+assert_logs_contain() {
+    local message="$1"
+    # Note: with image-rs we get more that the default 1000 lines of logs
+    journalctl -x -t kata --since "$start_date" -n 100000 | grep "$message"
+}
+
+@test "$test_tag Test can pull an image as a raw block disk image to guest with dm-verity enabled" {
+    if [ "$SNAPSHOTTER" = "nydus" ]; then
+        EXPORT_MODE="image_block_with_verity" RUNTIMECLASS="$RUNTIMECLASS" SNAPSHOTTER="nydus" configure_remote_snapshotter
+        pod_config="$(new_pod_config "$image_unsigned_unprotected")"
+        echo $pod_config
+        create_test_pod "$pod_config"
+        remove_test_image "$image_unsigned_unprotected"
+    fi
+}
+
+@test "$test_tag Test can create two pods with pulling the image only once" {
+    if [ "$SNAPSHOTTER" = "nydus" ]; then
+        EXPORT_MODE="image_block_with_verity" RUNTIMECLASS="$RUNTIMECLASS" SNAPSHOTTER="nydus" configure_remote_snapshotter
+
+        pod_config_1="$(new_pod_config "$image_unsigned_unprotected" "1")"
+        echo $pod_config_1
+        create_test_pod $pod_config_1
+        pod_config_2="$(new_pod_config "$image_unsigned_unprotected" "2")"
+        echo $pod_config_2
+        create_test_pod $pod_config_2
+
+        pull_times=$(journalctl -g "PullImage \"$image_unsigned_protected\" with snapshotter nydus" | wc -l)
+        [ ${#pull_times[@]} -eq 1 ]
+        remove_test_image "$image_unsigned_unprotected"
+    fi
+}
+
+@test "$test_tag Test can pull an image inside the guest with remote-snapshotter" {
+    switch_image_service_offload on
+    if [ "$SNAPSHOTTER" = "nydus" ]; then
+        EXPORT_MODE="image_guest_pull" RUNTIMECLASS="$RUNTIMECLASS" SNAPSHOTTER="nydus" configure_remote_snapshotter
+        create_test_pod
+        remove_test_image "$image_unsigned_unprotected"
+    fi
+}
```

Comment:
I wonder if longer term we might want to separate this out into a separate bats file (potentially with the existing `agent_image.bats` tests that do image pull on the guest and then the snapshotter configuration could move into the setup? We can think about this on the merge into `main` though, so it's not a blocker here.

Reply:
Sure. I'll update.

Reply:
At the moment, I believe segregating the snapshotter tests into a separate file can mitigate potential interference between snapshotter-related tests and the tests in `agent_image.bats`. However, I agree with you to separate tests of guest pulling out into a separate bats file and we can do it after that. What do you think?

Reply:
Yes - I'm happy with doing the refactoring of test locations after this set of PRs has gone in